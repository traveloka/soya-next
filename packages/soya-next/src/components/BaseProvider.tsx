import React from "react";
import { Cookies, CookiesProvider } from "react-cookie";
import LocaleProvider from "./LocaleProvider";
import type { SoyaNextLocaleContext } from "../types";

export interface BaseProviderProps extends SoyaNextLocaleContext {
  children: React.ReactNode;
  cookies?: Cookies;
}

class BaseProvider extends React.Component<BaseProviderProps> {
  render() {
    const {
      cookies,
      defaultLocale,
      siteLocales,
      locale,
      children,
    } = this.props;
    return (
      <CookiesProvider cookies={cookies}>
        <LocaleProvider
          defaultLocale={defaultLocale}
          siteLocales={siteLocales}
          locale={locale}
        >
          {React.Children.only(children)}
        </LocaleProvider>
      </CookiesProvider>
    );
  }
}

export default BaseProvider;
