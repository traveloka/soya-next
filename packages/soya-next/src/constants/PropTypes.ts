import PropTypes from "prop-types";

export const localeShape = PropTypes.shape({
  country: PropTypes.string,
  language: PropTypes.string,
});
