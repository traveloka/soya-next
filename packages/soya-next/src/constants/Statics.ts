export const NEXT_STATICS: Record<string, true> = {
  getInitialProps: true,
};
