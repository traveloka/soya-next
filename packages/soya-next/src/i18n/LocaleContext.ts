import React from "react";
import type { SoyaNextLocaleContext } from "../types";

const LocaleContext = React.createContext<SoyaNextLocaleContext>({
  defaultLocale: "",
  siteLocales: [],
  locale: null,
});

LocaleContext.displayName = "LocaleContext";

export default LocaleContext;
