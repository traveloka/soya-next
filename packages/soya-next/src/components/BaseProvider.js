import React from "react";
import PropTypes from "prop-types";
import { Cookies, CookiesProvider } from "react-cookie";
import { localeShape } from "../constants/PropTypes";
import LocaleProvider from "../components/LocaleProvider";

class BaseProvider extends React.Component {
  static propTypes = {
    cookies: PropTypes.instanceOf(Cookies).isRequired,
    defaultLocale: PropTypes.string,
    siteLocales: PropTypes.arrayOf(PropTypes.string.isRequired),
    locale: localeShape,
    children: PropTypes.element.isRequired
  };

  render() {
    const {
      cookies,
      defaultLocale,
      siteLocales,
      locale,
      children
    } = this.props;
    return (
      <CookiesProvider cookies={cookies}>
        <LocaleProvider
          defaultLocale={defaultLocale}
          siteLocales={siteLocales}
          locale={locale}
        >
          {React.Children.only(children)}
        </LocaleProvider>
      </CookiesProvider>
    );
  }
}

export default BaseProvider;
