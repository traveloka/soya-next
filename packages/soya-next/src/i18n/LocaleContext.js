import React from "react";
import PropTypes from "prop-types";
import { localeShape } from "../constants/PropTypes";

const LocaleContext = React.createContext({
  defaultLocale: "",
  siteLocales: [],
  locale: null
});

LocaleContext.Provider.displayName = "LocaleContext.Provider";
LocaleContext.Consumer.displayName = "LocaleContext.Consumer";

LocaleContext.Provider.propTypes = {
  value: PropTypes.shape({
    defaultLocale: PropTypes.string,
    siteLocales: PropTypes.arrayOf(PropTypes.string),
    locale: localeShape
  })
};

export default LocaleContext;
