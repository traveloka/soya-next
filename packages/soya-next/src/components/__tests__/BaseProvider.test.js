import React from "react";
import PropTypes from "prop-types";
import TestUtils from "react-dom/test-utils";
import { Cookies } from "react-cookie";
import LocaleContext from "../../i18n/LocaleContext";
import BaseProvider from "../BaseProvider";
import { withCookies } from "react-cookie";
import { localeShape } from "../../constants/PropTypes";

describe("<BaseProvider />", () => {
  let rootContext;
  const ContextChecker = withCookies(({ cookies }) => (
    <LocaleContext.Consumer>
      {context => ((rootContext = ({ ...context, cookies })), null)}
    </LocaleContext.Consumer>
  ));

  afterEach(() => {
    rootContext = undefined;
  });

  it("should add cookies, defaultLocale, locale, and siteLocales to child context", () => {
    const tree = TestUtils.renderIntoDocument(
      <BaseProvider
        cookies={new Cookies()}
        defaultLocale="id-id"
        locale={{ language: "en", country: "id" }}
        siteLocales={["id-id", "en-id"]}
      >
        <ContextChecker />
      </BaseProvider>
    );
    expect(rootContext).toMatchSnapshot();
  });
});
