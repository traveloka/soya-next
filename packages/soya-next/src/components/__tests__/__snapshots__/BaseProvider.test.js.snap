// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseProvider /> should add cookies, defaultLocale, locale, and siteLocales to child context 1`] = `
Object {
  "cookies": Cookies {
    "HAS_DOCUMENT_COOKIE": true,
    "changeListeners": Array [
      [Function],
    ],
    "cookies": Object {},
  },
  "defaultLocale": "id-id",
  "locale": Object {
    "country": "id",
    "language": "en",
  },
  "siteLocales": Array [
    "id-id",
    "en-id",
  ],
}
`;
